modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -294,14 +294,27 @@ public ObjectEntry addObjectEntry(
 			objectEntry.getUserId(), objectDefinition.getClassName(),
 			objectEntry.getPrimaryKey(), false, false, false);
 
+		boolean clearObjectEntryIdsMap =
+			ObjectActionThreadLocal.isClearObjectEntryIdsMap();
+
 		try {
+			if (clearObjectEntryIdsMap) {
+				ObjectActionThreadLocal.clearObjectEntryIdsMap();
+			}
+
+			ObjectActionThreadLocal.setClearObjectEntryIdsMap(false);
+
 			if (workflowAction == WorkflowConstants.ACTION_SAVE_DRAFT) {
 				ObjectEntryThreadLocal.setSkipObjectValidationRules(true);
 			}
 
 			objectEntry = objectEntryPersistence.update(objectEntry);
 		}
 		finally {
+			if (clearObjectEntryIdsMap) {
+				ObjectActionThreadLocal.setClearObjectEntryIdsMap(true);
+			}
```

Comment:
@devpedrotavares can we change to `ObjectActionThreadLocal.setClearObjectEntryIdsMap(clearObjectEntryIdsMap);`?